jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java

```diff
@@ -45,8 +45,32 @@
     public Runnable offer(Runnable task)
     {
         if (task == null)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Offering task null, skipping it in {}", this);
             return null;
+        }
+        if (NamedRunnable.LOG.isDebugEnabled())
+        {
+            if (!(task instanceof NamedRunnable))
+            {
+                StackTraceElement[] stackTrace = new Exception().getStackTrace();
+                for (StackTraceElement stackTraceElement : stackTrace)
+                {
+                    String className = stackTraceElement.getClassName();
+                    if (!className.equals(SerializedInvoker.class.getName()) && !className.equals(getClass().getName()))
+                    {
+                        String name = "Queued at " + className + "." + stackTraceElement.getMethodName() +
+                            "(" + stackTraceElement.getFileName() + ":" + stackTraceElement.getLineNumber() + ")";
+                        task = new NamedRunnable(task, name);
```

Comment:
Why are we setting the input param of the method during an isDebugEnabled?

Reply:
We're overwriting the input param when debug is enabled. The overwriting task has a nicely-readable-for-humans `toString()` but it absurdly costly to compute as it needs to take a stack trace then walk it, so this is only done when debug is enabled.

I should have explained that in a comment, let me add one.